digest/src/core_api/ct_variable.rs

```diff
@@ -165,3 +166,25 @@ where
         write!(f, "{}", OutSize::USIZE)
     }
 }
+
+impl<T, OutSize> SerializableState for CtVariableCoreWrapper<T, OutSize>
+where
+    T: VariableOutputCore + SerializableState,
+    OutSize: ArrayLength<u8> + IsLessOrEqual<T::OutputSize>,
+    LeEq<OutSize, T::OutputSize>: NonZero,
+    T::BlockSize: IsLess<U256>,
+    Le<T::BlockSize, U256>: NonZero,
+{
+    type SerializedStateSize = T::SerializedStateSize;
+
+    fn serialize(&self) -> SerializedState<Self> {
+        self.inner.serialize()
+    }
```

Comment:
As a sanity check it could be worth to add into serialized state output size (the `OutSize` type parameter) and return an error during deserialization if it does not match.